### DISCUSSION TITLE
Staking and gas tokens post warp

### DISCUSSION THREAD

**Discussion by minghinmatthewlam** [ava-labs/avalanchego#1912] [General]
Following questions are all in the context of warp already being launched:

1. When subnets are created or transition into elastic subnets, how can they use erc20 tokens are the staking token? After the erc20 token gets created on the C chain:
  - What features are needed to support the erc20 token being sent from C chain to the P chain to use as the staking token?
  - Does warp play a part in the transfer of erc20 from C to P? If so, what would the payload of the warp message include i.e. contract address?
2. Will subnets be able to use erc20 tokens created on the C chain as their native tokens for paying gas fees? If so, at subnet creation how will the erc20 token be transferred from C chain to subnet, and set as the native fee token? Will there be a warp protocol involved?
3. For subnet staking, do validators only need to connect to a P chain rpc endpoint to stake with multiple subnets? Is it correct for Cosmos staking, you'd need to connect to individual zones?

Thanks

**Comment by meaghanfitzgerald** [comment]
In 2022, a community member wrote a tutorial for the tutorials contest on [how to use an ERC-20 as the gas token](https://docs.avax.network/deprecated/tutorials-contest/2022/erc20-as-subnet-gas-token). Info may be out of date, but it goes through how to deploy a bridge and a relayer, and bridge a C-chain ERC-20 to a subnet.

**Reply by meaghanfitzgerald** [reply]
At a quick glance it doesn't seem like the tutorial was actually finished, bc there is no info on how to actually use the token as the gas token.

**Comment by Kielbiczek** [comment]
ok
